### PR TITLE
Add `_codecs.encode` and `builtins.bytearray` to `_get_allowed_globals` to support bytes and bytearray serialization

### LINKED ISSUE #133163: `torch.load` with `weights_only=True` does not allow `bytes` or `bytearray`
### 🐛 Describe the bug

```python
import torch

torch.save(b'hello', '/tmp/dummy.pth')
torch.load('/tmp/dummy.pth', weights_only=False) # OK
torch.load('/tmp/dummy.pth', weights_only=True) # Error
# UnpicklingError: Weights only load failed. This file can still be loaded, to do so you have two options 
# 	(1) Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
# 	(2) Alternatively, to load with `weights_only=True` please check the recommended steps in the following error message.
# 	WeightsUnpickler error: Unsupported global: GLOBAL _codecs.encode was not an allowed global by default. Please use `torch.serialization.add_safe_globals([encode])` to allowlist this global if you trust this class/function.

# Check the documentation of torch.load to learn more about types accepted by default with weights_only https://pytorch.org/docs/stable/generated/torch.load.html.

torch.save(bytearray(b'hello'), '/tmp/dummy.pth')
torch.load('/tmp/dummy.pth', weights_only=False) # OK
torch.load('/tmp/dummy.pth', weights_only=True) # Error, same as above
```

Adding `torch.serialization.add_safe_globals([bytes, bytearray])` at the beginning doesn't help.

The doc for `torch.load` says:
```plaintext
        weights_only: Indicates whether unpickler should be restricted to
            loading only tensors, primitive types, dictionaries
            and any types added via :func:`torch.serialization.add_safe_globals`.
```

I think `bytes` and `bytearray` are primitive types and should be supported.

Possibly related: #118092

### Versions

`torch==2.5.0.dev20240810+cpu`

cc @mruberry @mikaylagawarecki

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #133163

Debugged in collaboration with @hariveliki 

The `byte` type is demanding the global `_codecs.encode`. That means, the following currently works:
```python
import torch

torch.save(b'hello', '/tmp/dummy.pth')

torch.serialization.add_safe_globals([_codecs.encode])
torch.load('/tmp/dummy.pth', weights_only=True)
```

Similarly, `bytearray` needs `builtins.bytearray`.

Following the `torch.loads` docs promise, both types should be supported without `add_safe_globals` as they are both primitive types:
>         weights_only: Indicates whether unpickler should be restricted to
>            loading only tensors, primitive types, dictionaries
>           and any types added via :func:`torch.serialization.add_safe_globals`.

This PR adds both `_codecs.encode` and `builtins.bytearray` to `_get_allowed_globals` and test for saving and loading of both types with and without `weights_only`.

### CODE REVIEW

File: test/test_serialization.py [hunk 1]
```diff
@@ -4133,6 +4133,17 @@ def test_serialization_dtype(self, dtype, weights_only):
             y['even'][0] = torch.tensor(-0.25, dtype=dtype)
             self.assertEqual(y['x'][:2].to(dtype=torch.float32), torch.tensor([-0.25, 0.25]))
 
+    @parametrize('byte_literals', (b'byte', bytearray(b'bytearray')))
+    @parametrize('weights_only', (True, False))
+    def test_serialization_byte_literal(self, byte_literals, weights_only):
+        """ Tests that by literal can be serialized.
```
Comment (mikaylagawarecki) [inline test/test_serialization.py:4139]: nit: 
```suggestion
        """ Tests that byte literal can be serialized.
```

**Review by mikaylagawarecki — APPROVED** [review]
Thanks, will merge if CI is green 

edit: please fix lint

**Comment by 1ucian0** [comment]
Thanks @mikaylagawarecki  for the review!

fixes:
 * typo in the test docstring
 * lint

**Pull request closed without merging**